include/lbann/execution_contexts/execution_context.hpp

```diff
@@ -105,6 +107,21 @@ class execution_context {
     return const_cast<trainer&>(static_cast<const execution_context&>(*this).get_trainer());
   }
 
+  inline void set_training_algorithm(training_algorithm* t){
```

Comment:
Should you be able to explicitly set the training algorithm to `nullptr`? If not, this should take a reference.